pkg/api/http_server.go

```diff
@@ -58,6 +59,7 @@ func (hs *HttpServer) Start(ctx context.Context) error {
 
 	listenAddr := fmt.Sprintf("%s:%s", setting.HttpAddr, setting.HttpPort)
 	hs.log.Info("Initializing HTTP Server", "address", listenAddr, "protocol", setting.Protocol, "subUrl", setting.AppSubUrl, "socket", setting.SocketPath)
+	util.SdNotify("READY=1")
```

Comment:
Not sure I think this is the best place. Would prefer to handle all process related code in https://github.com/grafana/grafana/blob/master/pkg/cmd/grafana-server/server.go perhaps at https://github.com/grafana/grafana/blob/master/pkg/cmd/grafana-server/server.go#L98

Technically we are still not listening to the port yet but the code between that point and where you have added it should be very quick. Would that be a problem?

Reply:
No, not at all. The closer to when the service is actually listening, the better. I'll move it over there.  I only picked the spot I picked because that's where the "Initializing HTTP Server" message gets logged.

Reply:
Oh, I misunderstood. The file you pointed me to is actually further from where the listener gets created. Should still be ok if there are no major amounts of I/O in between (writing out log messages is fine, database migrations are a bit much). Let me have a look at what happens in between...

Reply:
Ok, I moved this and checked it against the chef recipe that used to run into this race condition - works just fine.